### PR TITLE
Add sprockets-es6 for .es6 support in asset pipeline

### PR DESCRIPTION
**Pull request by srih4ri**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,15 +7,15 @@ gem 'rails', '4.2.1'
 gem 'arel'
 gem 'jquery-rails'
 
-gem 'sprockets-rails', '3.0.0.beta1'
```
Comment (vipulnsward) [inline Gemfile]: Why is this needed? Last I remember, latest sprockets-rails is actually needed for `sprockets-es6`

Comment (srih4ri) [inline Gemfile]:
``` 
-gem 'sprockets-rails', '3.0.0.beta1'
+gem 'sprockets-rails'
```
should include the latest `sprockets-rails`, right?

File: Gemfile.lock [hunk 2]
```diff
@@ -264,17 +267,15 @@ GEM
...
-    sprockets-rails (3.0.0.beta1)
-      actionpack (>= 4.0)
-      activesupport (>= 4.0)
-      sprockets (~> 2.8)
-    sprockets-strict-mode (1.0)
-      sprockets
+    sprockets-es6 (0.6.2)
+      babel-transpiler
+      sprockets (>= 3.0.0)
+    sprockets-rails (2.3.2)
```
Comment (vipulnsward) [inline Gemfile.lock]: Without above, `sprockets-rails` actually picks the stable version instead of beta.

Comment (srih4ri) [inline Gemfile.lock]: When I update it to : `gem 'sprockets-rails', '>=3.0.0.beta1'` and run bundle install, I run into this :

```
~/c/wheel (sprockets-es6 ⚡) bundle
Fetching gem metadata from https://rubygems.org/........
Fetching version metadata from https://rubygems.org/...
Fetching dependency metadata from https://rubygems.org/..
You have requested:
  sprockets-rails >= 3.0.0.beta1

The bundle currently has sprockets-rails locked at 2.3.2.
Try running `bundle update sprockets-rails`
~/c/wheel (sprockets-es6 ⚡) bundle update sprockets-rails
Fetching gem metadata from https://rubygems.org/........
Fetching version metadata from https://rubygems.org/...
Fetching dependency metadata from https://rubygems.org/..
Resolving dependencies........
Bundler could not find compatible versions for gem "sprockets":
  In Gemfile:
    sprockets (< 4.0, >= 2.8) ruby

    sprockets-es6 (>= 0) ruby depends on
      sprockets (>= 3.0.0) ruby

    sprockets-rails (>= 3.0.0.beta1) ruby depends on
      sprockets (~> 2.8) ruby

```
